src/main/java/com/exasol/adapter/document/files/FileTypeSpecificSchemaFetcher.java

```diff
@@ -0,0 +1,73 @@
+package com.exasol.adapter.document.files;
+
+import java.util.Optional;
+
+import com.exasol.adapter.document.documentfetcher.files.RemoteFile;
+import com.exasol.adapter.document.documentfetcher.files.RemoteFileFinder;
+import com.exasol.adapter.document.edml.MappingDefinition;
+import com.exasol.adapter.document.iterators.CloseableIterator;
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * Interface for file type specific mapping-fetcher implementations.
+ */
+@FunctionalInterface
+public interface FileTypeSpecificSchemaFetcher {
+
+    /**
+     * Detect the schema of the given file or files.
+     *
+     * @param fileFinder the file finder providing the files for which to detect the mapping.
+     * @return an empty {@link Optional} if this file type does not support schema detection or the
+     *         {@link MappingDefinition}.
+     */
+    Optional<MappingDefinition> fetchSchema(RemoteFileFinder fileFinder);
+
+    /**
+     * Create a new "unsupported" {@link FileTypeSpecificSchemaFetcher} that always returns an empty {@link Optional}.
```

Comment:
```suggestion
     * Create a new {@link FileTypeSpecificSchemaFetcher} that always returns an empty {@link Optional}.
```

Reply:
Maybe also makes sense to rename the method to `empty()` instead of `unsupported()`?